lib/iris/cube.py

```diff
@@ -2293,10 +2293,23 @@ def cell_methods(self):
         return self._metadata_manager.cell_methods
 
     @cell_methods.setter
-    def cell_methods(self, cell_methods):
-        self._metadata_manager.cell_methods = (
-            tuple(cell_methods) if cell_methods else tuple()
-        )
+    def cell_methods(self, cell_methods: Iterable):
+        if not cell_methods:
+            # For backwards compatibility: Empty or null value is equivalent to ().
+            cell_methods = ()
+        else:
+            # Can supply any iterable, which is converted (copied) to a tuple.
+            cell_methods = tuple(cell_methods)
+            for cell_method in cell_methods:
+                # All contents should be CellMethods.  Requiring class membership is
+                # somewhat non-Pythonic, but simple, and not a problem for now.
+                if not isinstance(cell_method, iris.coords.CellMethod):
+                    msg = (
+                        f"Cube.cell_methods assigned value includes {cell_method}, "
+                        "which is not an iris.coords.CellMethod."
+                    )
+                    raise ValueError(msg)
```

Comment:
Assumedly we could attempt to handle the incorrectly-typed object here rather than immediately erroring? That way we avoid crashing out of the load at the first sign of failure.

For example, we could instead:
* ignore the not-a-`CellMethod` and not include it in the final list of cell methods. This might mean we don't fully represent all of the averaging metadata of the incoming dataset, of course, which is undesirable - but if the object is already malformed maybe we're better off skipping it anyway?
* try passing the not-a-`CellMethod` back through the `CellMethod` constructor and re-inserting it into the list of cell methods. That way if there is a problem with the averaging metadata we can hopefully fail within the `CellMethod` constructor with a more accurate description of _why_ the object is causing the problem.

Reply:
> objects that aren't cell methods that show up in the list of cell methods

I don't quite get this.  The change here only operates when you assign something to 'cube.cell_methods', so I can't see why we would want to attempt anything _except_ to say that we only accept CellMethod instances.

My thinking is, I don't think this would ever be triggered within other Iris operations, it would only happen in a user-code assignment.

Reply:
I'm happy with this. The main thing is that we're confident the various loaders have already simplified things to either `None` or a valid `CellMethod` - nothing else is expected to get this far unless a user is explicitly using the `setter`.

And in all cases I would be uncomfortable with attempting a conversion to a valid `CellMethod` object, as this is fairly opaque behaviour that separates the user from the true situation (often referred to as "magic").